libraries/AP_Proximity/AP_Proximity_LightWareSF40C.h

```diff
@@ -19,78 +20,124 @@ class AP_Proximity_LightWareSF40C : public AP_Proximity_Backend
+    // request start of streaming of distances
+    void request_stream_start();
+
+    // request token of sensor (required for reset)
+    void request_token();
+    bool got_token() const { return (sensor_state.token[0] != 0 || sensor_state.token[1] != 0); }
+    void clear_token() { memset(sensor_state.token, 0, ARRAY_SIZE(sensor_state.token)); }
+
+    // request reset of sensor
+    void request_reset();
 
     // check and process replies from sensor
-    bool check_for_reply();
-    bool process_reply();
-    void clear_buffers();
-
-    // reply related variables
-    AP_HAL::UARTDriver *uart = nullptr;
-    char element_buf[2][10];
-    uint8_t element_len[2];
-    uint8_t element_num;
-    bool ignore_reply;                      // true if we should ignore the incoming message (because it is just echoing our command)
-    bool wait_for_space;                    // space marks the start of returned data
-
-    // request related variables
-    enum RequestType _last_request_type;    // last request made to sensor
-    uint8_t  _last_sector;                  // last sector requested
-    uint32_t _last_request_ms;              // system time of last request
-    uint32_t _last_distance_received_ms;    // system time of last distance measurement received from sensor
-    uint8_t _request_count;                 // counter used to interleave requests for distance with health requests
-
-    // sensor health register
-    union {
-        struct PACKED {
-            uint16_t motor_stopped : 1;
-            uint16_t motor_dir : 1;          // 0 = clockwise, 1 = counter-clockwise
-            uint16_t motor_fault : 1;
-            uint16_t torque_control : 1;     // 0 = automatic, 1 = manual
-            uint16_t laser_fault : 1;
-            uint16_t low_battery : 1;
-            uint16_t flat_battery : 1;
-            uint16_t system_restarting : 1;
-            uint16_t no_results_available : 1;
-            uint16_t power_saving : 1;
-            uint16_t user_flag1 : 1;
-            uint16_t user_flag2 : 1;
-            uint16_t unused1 : 1;
-            uint16_t unused2 : 1;
-            uint16_t spare_input : 1;
-            uint16_t major_system_abnormal : 1;
-        } _flags;
-        uint16_t value;
-    } _sensor_status;
-
-    // sensor config
-    uint8_t _motor_speed;               // motor speed as reported by lidar
-    uint8_t _motor_direction = 99;      // motor direction as reported by lidar
-    int16_t _forward_direction = 999;   // forward direction as reported by lidar
-    bool _sector_initialised = false;
+    void process_replies();
+
+    // process one byte received on serial port
+    // state is stored in msg structure.  when a full package is received process_message is called
+    void parse_byte(uint8_t b);
+
+    // process the latest message held in the msg structure
+    void process_message();
+
+    // internal variables
+    AP_HAL::UARTDriver *uart;               // uart for communicating with sensor
```

Comment:
In the original source, "_" was added to the beginning of the variable for the class variable.
I could easily recognize that “_” was a class variable.

Reply:
ok, reverted to putting an underscore before all the variable names.